packages/aws-cdk-lib/aws-appconfig/test/configuration.test.ts

```diff
@@ -283,6 +283,153 @@ describe('configuration', () => {
     Template.fromStack(stack).resourceCountIs('AWS::AppConfig::Deployment', 2);
   });
 
+  test('configuration with retrievalRole undefined from bucket source should create a new role', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const app = new Application(stack, 'MyAppConfig', {
+      applicationName: 'MyApplication',
+    });
+    const bucket = new Bucket(stack, 'MyBucket');
+
+    // WHEN
+    new SourcedConfiguration(stack, 'MySourcedConfig', {
+      versionNumber: '1',
+      location: ConfigurationSource.fromBucket(bucket, 'path/to/object'),
+      application: app,
+    });
+
+    // THEN
+    // should have a new role provisioned with AllowAppConfigReadFromSourcePolicy
+    Template.fromStack(stack).hasResourceProperties('AWS::IAM::Role', {
+      AssumeRolePolicyDocument: {
+        Statement: [
+          {
+            Action: 'sts:AssumeRole',
+            Effect: 'Allow',
+            Principal: {
+              Service: 'appconfig.amazonaws.com',
+            },
+          },
+        ],
+        Version: '2012-10-17',
+      },
+      Policies: [
+        {
+          PolicyName: 'AllowAppConfigReadFromSourcePolicy',
+        },
+      ],
+    });
+  });
+
+  test('configuration with retrievalRole defined should NOT create a new role', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const app = new Application(stack, 'MyAppConfig', {
+      applicationName: 'MyApplication',
+    });
+    const bucket = new Bucket(stack, 'MyBucket');
+
+    // WHEN
+    new SourcedConfiguration(stack, 'MySourcedConfig', {
+      versionNumber: '1',
+      location: ConfigurationSource.fromBucket(bucket, 'path/to/object'),
+      application: app,
+      retrievalRole: new iam.Role(stack, 'MyRole', {
+        assumedBy: new iam.ServicePrincipal('appconfig.amazonaws.com'),
+      }),
+    });
+
+    // THEN
+    // should have a new role provisioned with AllowAppConfigReadFromSourcePolicy
```

Comment:
Is this comment correct? Or should it be:
```ts
// should NOT have a new role provisioned with AllowAppConfigReadFromSourcePolicy
```

Reply:
Yes you are right! Thank you. Just fixed.